src/gui/torrentcreatordialog.cpp

```diff
@@ -90,13 +90,11 @@ TorrentCreatorDialog::TorrentCreatorDialog(QWidget *parent, const Path &defaultP
     connect(m_ui->buttonCalcTotalPieces, &QPushButton::clicked, this, &TorrentCreatorDialog::updatePiecesCount);
     connect(m_ui->checkStartSeeding, &QCheckBox::clicked, m_ui->checkIgnoreShareLimits, &QWidget::setEnabled);
 
-    connect(m_creatorThread, &BitTorrent::TorrentCreatorThread::creationSuccess, this, &TorrentCreatorDialog::handleCreationSuccess);
-    connect(m_creatorThread, &BitTorrent::TorrentCreatorThread::creationFailure, this, &TorrentCreatorDialog::handleCreationFailure);
-    connect(m_creatorThread, &BitTorrent::TorrentCreatorThread::updateProgress, this, &TorrentCreatorDialog::updateProgressBar);
-
     loadSettings();
     updateInputPath(defaultPath);
 
+    m_threadPool.setMaxThreadCount(1);
```

Comment:
Doesn't that make these changes completely meaningless? So you could continue to use the previous (QThread based) implementation.

Reply:
> what was your conclusion?

Making "torrent creator" compatible with the ability to run under QThreadPool is generally a good idea. "Torrent creator" dialog is just a special (degenerate) case at the moment.